### PR TITLE
Additional datastore change log since 0.10.0

### PR DESCRIPTION
**Pull request by wooj2**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ A declarative library for application development using cloud services.
   - Migrate APIPlugin's websocket provider to use AppSyncRealTimeClient [#330](https://github.com/aws-amplify/amplify-ios/pull/330), [#341](https://github.com/aws-amplify/amplify-ios/pull/341), [#352](https://github.com/aws-amplify/amplify-ios/pull/352)
 - **DataStore**
   - DataStore.save() API now takes in a condition [#355](https://github.com/aws-amplify/amplify-ios/pull/355)
+  - Added reachability and retryability to remote sync engine [#321](https://github.com/aws-amplify/amplify-ios/pull/321) [#322](https://github.com/aws-amplify/amplify-ios/pull/322) [#323](https://github.com/aws-amplify/amplify-ios/pull/323) [#324](https://github.com/aws-amplify/amplify-ios/pull/324)
+  - Fix out of bounds case when retry handler >= 57 [#338](https://github.com/aws-amplify/amplify-ios/pull/338)
+  - Datastore.delete() with predicate [#346](https://github.com/aws-amplify/amplify-ios/pull/346)
```
Comment (lawmicha) [inline CHANGELOG.md:21]: `- Datastore.delete(modelType) with predicate [#346](https://github.com/aws-amplify/amplify-ios/pull/346)`

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,10 @@ A declarative library for application development using cloud services.
   - Migrate APIPlugin's websocket provider to use AppSyncRealTimeClient [#330](https://github.com/aws-amplify/amplify-ios/pull/330), [#341](https://github.com/aws-amplify/amplify-ios/pull/341), [#352](https://github.com/aws-amplify/amplify-ios/pull/352)
 - **DataStore**
   - DataStore.save() API now takes in a condition [#355](https://github.com/aws-amplify/amplify-ios/pull/355)
```
Comment (lawmicha) [inline CHANGELOG.md:18]: `- DataStore.save() with condition [#355](https://github.com/aws-amplify/amplify-ios/pull/355)`

for consisent wording

Comment (wooj2) [reply CHANGELOG.md]: I'm assuming you wanted me to overwrite the changes you recently submitted? if so.. I think we are now good to go

File: CHANGELOG.md (hunk 1 above)
Comment (wooj2) [reply CHANGELOG.md]: updated! thanks!

**Pull request merged**